src/uct/cuda/cuda_ipc/cuda_ipc_iface.c

```diff
@@ -171,6 +218,61 @@ static unsigned uct_cuda_ipc_iface_progress(uct_iface_h tl_iface)
                                          max_events);
 }
 
+static ucs_status_t uct_cuda_ipc_iface_event_fd_arm(uct_iface_h tl_iface,
+                                                    unsigned events)
+{
+    uct_cuda_ipc_iface_t *iface = ucs_derived_of(tl_iface, uct_cuda_ipc_iface_t);
+    unsigned max_events         = iface->config.max_poll;
+    int ret;
+    int i;
+    uint64_t dummy;
+    ucs_status_t status;
+
+    if (uct_cuda_ipc_progress_event_q(iface, &iface->outstanding_d2d_event_q,
+                                      max_events)) {
+        return UCS_ERR_BUSY;
+    }
+
+    do {
+        ret = read(iface->eventfd, &dummy, sizeof(dummy));
+        if (ret == sizeof(dummy)) {
+            status = UCS_ERR_BUSY;
+            return status;
+        } else if (ret == -1) {
+            if (errno == EAGAIN) {
+                break;
+            } else if (errno != EINTR) {
+                ucs_error("Read from internal event fd failed: %m");
+                status = UCS_ERR_IO_ERROR;
+                return status;
+            } else {
+                return UCS_ERR_BUSY;
+            }
+        } else {
+            ucs_assert(ret == 0);
+            goto add_cbs;
```

Comment:
why need this goto?

Reply:
We had more than one case where add_cbs was needed. We don't anymore. I'm removing this.